Celeste.Mod.mm/Mod/UI/OuiModOptionString.cs

```diff
@@ -183,7 +184,8 @@ public OuiModOptionString()
         }
 
         public override IEnumerator Leave(Oui next) {
-            TextInput.OnInput -= OnTextInput;
+            if (UseKeyboardInput) 
```

Comment:
I'd prefer for these (all the Ouis that follow this pattern) to check whether the event was added instead of checking the setting value in case someone decides to change the value of the setting at a weird moment.
Also, this would've been a thing before as well, but what if we're exiting the overworld while text input is active (e.g. via a console command run by a TAS)? Are we leaking the subscription in that case?